### ISSUE TITLE
20mm-30mm script causes very bad server desynch.

### ISSUE THREAD

**Issue by Jester814** [KoffeinFlummi/AGM#2152]
My unit has been using the RHS A-10 and the vanilla A-164 and every time the main gun fires the server goes into a desync period and has even crashed in the middle of an operation on more than one occasion.

We finally nailed down the problem to the script that changes 20mm rounds to 30mm rounds when the GAU is fired. It causes massive RPT spam issues as well as the server desynch:

17:22:12 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:12 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:13 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:14 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:14 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:14 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:14 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:42 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:43 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:43 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:43 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:43 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:43 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:43 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:43 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:44 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:45 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:45 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:45 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:45 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:45 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:45 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:45 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy
17:22:45 Submunition: AGM_Gatling_30mm_HE_Plane_CAS_01_Deploy

With a server that has 50-60 personnel on it this is a very bad problem. I'm not sure how bad it is on a less populated server.

**Comment by jonpas** [comment]
Is that firing with RHS' A-10 or Vanilla's A-164 Wipeout?

**Comment by FrostOX9** [comment]
It's firing with both the RHS A-10 and the A-164 wipeout. We tested it with both aircraft.

**Comment by jonpas** [comment]
Was that happening before Arma 3 Game Update 1.40?

**Comment by FrostOX9** [comment]
Yes. The date of our first test was 22FEB. Our second test (which included the A-164) was on 1MAR.

**Comment by jonpas** [comment]
Probably not, but it might have been fixed in 1.40?

**Comment by esteldunedain** [comment]
@Jester814, Arma 3 is apparently writting to RPT each time a submunition is deployed. The GAU use submunition to allow for that high fire rate, so the log is spammed big time. There's not much we can do about it, besides temporarily removing the AGM_Aircraft.pbo.

 I hope they'll fix that soon.

**Comment by rlex** [comment]
>With a server that has 50-60 personnel on it this is a very bad problem. I'm not sure how bad it is on a less populated server.

Massive desync and lags even with 2-4 ppl on server.

**Comment by Cyruz143** [comment]
Has anyone logged a ticket for this that we can vote on for visibility?

**Comment by rlex** [comment]
will -nologs solve that as temporary workaround?

**Comment by esteldunedain** [comment]
Yes, probably; you're right.

The desync is caused exclusively by the time it takes the server to open the log file, write and close it again; Arma does this on the main code thread, thus blocking the excecusion. If the RPT in ont written then there shouldn't be any performance degradation.

**Comment by Cyruz143** [comment]
Anyone had a chance to try this on the perf client/server binary?

https://www.dropbox.com/sh/582opsto4mmr8d8/3BSy9PdRGm

**Comment by JamesTheClarke** [comment]
Is there an official BI Tracker open for the issues caused by BIS' logging evey sub-munition event to the RPT file? If yes, please post a link and I'll add my vote.

**Comment by Cyruz143** [comment]
Someone needs to test it on the perf binary, I asked Dwarden and he seemed to think it was some debug logging stuff that ended up in the main build by accident.
